src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -574,6 +586,44 @@ private static Details getDetailsForLambda(DetailAST ast) {
             return new Details(lcurly, rcurly, nextToken, shouldCheckLastRcurly);
         }
 
+        /**
+         * Collects details for Anonymous classes.
+         * @param ast a {@code DetailAST} value
+         * @return an object containing all details to make a validation
+         */
+        private static Details getDetailsForAnonymousClass(DetailAST ast) {
```

Comment:
The name of the method is not good. It checks whether the class is anonymous and if not, it returns default Details object (a king of null object).

Reply:
@MEZk Please suggest a satisfactory name for this method as well. Thanks.

Reply:
@ps-sp ,

to have good name we need to find a reason how it could be possible to have after NEW token nonAnonimous class. Please look at my comment below.